pkg/k8shandler/deployment.go

```diff
@@ -500,43 +500,44 @@ func (node *deploymentNode) isChanged() bool {
 	}
 
 	// Only Image and Resources (CPU & memory) differences trigger rolling restart
-	// we will only have one container, no need to do range
-	nodeContainer := node.self.Spec.Template.Spec.Containers[0]
-	desiredContainer := desired.Spec.Template.Spec.Containers[0]
+	for index := 0; index < len(node.self.Spec.Template.Spec.Containers); index++ {
+		nodeContainer := node.self.Spec.Template.Spec.Containers[index]
+		desiredContainer := desired.Spec.Template.Spec.Containers[index]
 
-	// check that both exist
+		// check that both exist
 
-	if nodeContainer.Image != desiredContainer.Image {
-		logrus.Debugf("Resource '%s' has different container image than desired", node.self.Name)
-		nodeContainer.Image = desiredContainer.Image
-		changed = true
-	}
+		if nodeContainer.Image != desiredContainer.Image {
```

Comment:
There looks like a refactoring opportunity to move this repeated code into a common function

Reply:
future tech debt?

Reply:
sure